app/views/users/show.html.erb

```diff
@@ -4,7 +4,9 @@
   </div>
 </div>
 
-<% if !@user.private? || @user == current_user || current_user.leaders.include?(@user) %>
+<% if @user == current_user ||
```

Comment:
try to keep involved logic like this in controller

Reply:
You should use Pundit here.
`<% if policy(@user).show? %>`